packages/graphile-build-pg/src/QueryBuilder.js

```diff
@@ -165,7 +165,7 @@ class QueryBuilder {
 
   // Helper function
   jsonbBuildObject(fields: Array<[SQL, RawAlias]>) {
-    if (this.supportsJSONB) {
+    if (this.supportsJSONB && fields.length > 50) {
```

Comment:
Fixes a performance regression for unindexed selects - takes query time from ~4s back to ~1.7s in this case.